### ISSUE TITLE
Need Version and Build Indication Display

### ISSUE THREAD

**Issue by PropGit** [parallaxinc/BlocklyProp#335]
@michellampo, @DarkmatterVale, @cbradyatinquire, @zfi, @kgracey 
We have a problem I'd like to see us solve swiftly.
### Problem

We desperately need the BlocklyProp interface to display version and build information that uniquely identifies the running instance that the user is using.  Currently, there's nothing except the change log (which is a different topic).  Such Version & Build information is vital for our development efforts as well.

Though I'd love for this information but be displayed right next to the BlocklyProp name in the upper-left corner of the pages...
![image](https://cloud.githubusercontent.com/assets/1266318/16738425/762c1cc2-474a-11e6-8a31-de42860748f4.png)

There's precious space up there that we should keep simple, so I'm suggesting this Version+Build information be kept in the footer, perhaps in the circled area here:
![image](https://cloud.githubusercontent.com/assets/1266318/16738314/0a4238ac-474a-11e6-8de3-152a34bfba20.png)

Which unfortunately means that information is not visible on the BlocklyProp Editor page; however, a simple click of the _Help_ menu will lead them to a page with the footer.
![image](https://cloud.githubusercontent.com/assets/1266318/16738560/0a575a92-474b-11e6-961c-58d6e235f9ee.png)
### How Do We Format and Implement Version?

I propose that the version information contain a numerical _major_ and _minor_ value that is manually set by us, and also a _build_ number that is automatically incremented by our Jenkins CI system as well as the first 7 digits of the BlocklyProp commit hash that it was generated from.  This could appear as something like this in the designated area:

v **m**.**nn** (**bbbbb**-**hhhhhhh**)

where **m** = major version, **n** = minor version, **b** = build number, **h** = commit hash.

For example: v1.15 (28374-f4ec148)

We can add a file (called version.txt) to the demo and master branches which contains our manually set major.minor value (ie: "1.15") and Jenkins can automatically create modified output that contains this plus the auto-generated build number and the BlocklyProp commit hash involved.

@zfi - I'm assuming Jenkins CI can output this in the form of a file included in the .war file.

@michellampo - The BlocklyProp system would have to be enhanced to find this generated information and display it on the footer of the site in the format above.
### What the Solution Does for Us
- This gives users an easy-to-find major/minor version number to reference.
- This also gives us developers a reference to draw upon for the actual BlocklyProp commit in use; good for 1) knowing when your push to the Demo System is live, 2) Giving a nice reference point for new issue postings, 3) seeing the build number which can change independent of the commit hash if a subsystem is changed without a BlocklyProp change.

What do you think?

**Comment by kgracey** [comment]
Great idea, Jeff. This would be helpful for us as well as we bounce between the demo.blockly.parallax.com and blocky.parallax.com, wondering about whether or not the two are the same version. - Ken

**Comment by DarkmatterVale** [comment]
I think this is a great idea. How do we decide when to change major and minor versions?

**Comment by PropGit** [comment]
###For when to change major and minor versions:
This can be a tricky thing to hande.  Normally I'd suggest that Demo drives the version forward and Production assumes that version when all of Demo is merged in; however, it's very likely we will rarely have all of Demo's current features merged in at once, which makes Production's version numbers only loosely connected to Demo's.  In light of this, I think we should do the following:

#### In General
- Major Version - Increment when significant or backwards incompatible system changes are included
- Minor Version - Increment when a set of bug fixes and backwards compatible system changes are included

#### On Production Releases
- Increment according to above rules when merging in a set of features/fixes
- Adjust _demo_ branch's version immediately after
- Create a release point in BlocklyProp repo that reflects the version and delivers a bulleted list of all the changes (in layman's terms) that link to top related issue for each.

#### On Demo Releases
- Increment version (Major.Minor+1) to be ahead of Production Release as soon as a new Production Release is made live
  - This is because Demo's intent is always to test new features/fixes not yet in Production
- Otherwise, leave version (Major.Minor) as-is (as we submit more commits to the demo branch) and let the build-system auto-adjust the build number and commit hash as an identifying factor

**Comment by paragitadmin** [comment]
This is partially implemented in the current build system.
